### DISCUSSION TITLE
NotHandledResponse obsolete

### DISCUSSION THREAD

**Discussion by RethaLam** [microsoft/Dynamics365Commerce.Solutions#29] [Q&A] [answered]
I am in the process to change our custom over from RetailSDK to CommerceSDK.  I watched the webinars and in one of them it stated that we can use NotHandledResponse() when we in certain instances want to run standard base code in our CRT custom code.
However when I use it, it is underlined with a green line and when I put my cursor on it, it gives me: "CS0618: notHandledResponse.NotHandeldResponse() is obsolete. Use static "instance' property instead. Obsolete since 10.8"

Now what does that mean? It is used in many places in the Microsoft documents under Learn.Microsoft.com.

**Comment by madyke** [comment, marked as answer]
Hi @RethaLam - that just means that the old method `NotHandledResponse()` has been deprecated, and that users should now use `NotHandledResponse.Instance` instead. This `Instance` property accomplishes all of the same things as the previous method, but uses a singleton to avoid unneeded allocations, improving performance.

As we make changes to classes like this, we will mark the old methods as 'Obsolete', meaning customers will see a warning like this one, which specifies when the method was deprecated. Generally though, the old 'Obsolete' methods remain available for customers to use for up to one year from when they were deprecated, in order to give them time to make the transition. Sometime after that grace period, we will remove the method so customers will not be able to call it anymore.

We try to keep documentation up to date as much as possible when methods or properties are deprecated, but it's always possible we miss some instances or changes. I'll file a bug for the documentation team to review the docs to see where we need to remove the obsolete method.

**Comment by RethaLam** [comment]
Hi @Madyke

I finally had a chance to get back to this code.  I was not able to find a reference to "NotHandledResponse.Instance" in the documentation.
I just downloaded the "Commerce home page" on Learn.Microsoft. A total of 6628 pages and NotHandledResponse is referenced 15 times. Not a single example uses it the way you suggested. All 15 occurrences are still just: NotHandledResponse();
Even the page that discussed: Executing the Next CRT handler - Chain of handlers.

![image](https://user-images.githubusercontent.com/123098075/224444261-d4d54342-a0a6-4916-a19d-ff7f507bf154.png)

I in the end searched for it in the code examples under RetailSDK and found it in the CommerceRuntime examples e.g.  GetSalesTransactionCustomReceiptFieldService.

I'm going to miss the examples that are under RetailSDK:
"RetailSDK\SampleExtensions"
"RetailSDK\POS\Extensions" especially SampleExtensions, SampleExtensions2, StoreHoursSample, SuspendTransactionReceiptSample and WarrantyAndReturnSample.
 It doesn't look like they were added to GitHub for commerceSDK which is unfortunate. They were a huge help when I had to taught myself RetailSDK development because of the lack of formal training available out there.
If they are all on gitHub for CommerceSDK, please provide the link because I can't find them.

**Reply by madyke** [reply]
Hi @RethaLam - I checked and the docs update bug I created for the documentation team to go and make the updates is still active and in their backlog, so they will get to when they have time based around their other priorities.

For the samples you mentioned, we have a backlog item to go back and transfer as many of the existing sample extensions as we can. Hopefully that will take place over the next couple of months, but for now we don't have a specific date for when they would be available.

**Comment by RethaLam** [comment]
Thank you Madyke
